backend/src/data/repositories/user-details/user-details.repository.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  UserDetailsUpdateImageRequestDto,
+  UserDetailsUpdateInfoRequestDto,
+} from '~/common/types/types';
+import { UserDetails as UserDetailsM } from '~/data/models/models';
+
+type Constructor = {
+  UserDetailsModel: typeof UserDetailsM;
+};
+
+class UserDetails {
+  #UserDetailsModel: typeof UserDetailsM;
+
+  public constructor({ UserDetailsModel }: Constructor) {
+    this.#UserDetailsModel = UserDetailsModel;
+  }
+
+  public async createUserDetails(
+    userId: number,
+    userDetails: UserDetailsUpdateInfoRequestDto,
+  ): Promise<UserDetailsM> {
+    const { lastName, firstName, gender, dateOfBirth } = userDetails;
+
+    return this.#UserDetailsModel.query().insert({
+      firstName,
+      lastName,
+      gender,
+      dateOfBirth,
+      userId,
+    });
```

Comment:
@PavelSukhinin could you explain to me please, why it's better to use execute() instead of async/await?

Reply:
@kalinkaaaa14 it's because objection is actually returning QueryBuilder instance. Please take a look [here](https://github.com/BinaryStudioAcademy/bsa-2022-guruhub/pull/103#discussion_r945107162)